tools/download-deps.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# shellcheck shell=bash
+
+set -eo pipefail
+
+unset CD_PATH
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+cd "${SCRIPT_DIR}" || exit 1
+
+. "${SCRIPT_DIR}/functions.sh"
+
+DEP_FILE=${1}
+TAG=${2:-$(${SCRIPT_DIR}/image-tag)}
+BIN_DIR=${3:-${SCRIPT_DIR}/bin}
+RELEASE_GOOS=${RELEASE_GOOS:-$(go env GOOS)}
+
+# Derive from GOOS
+RELEASE_OS=$(title_case "$RELEASE_GOOS")
+
+# create bin directory
+rm -rf "${BIN_DIR}"
+mkdir "${BIN_DIR}"
+
+# add binaries
+
+# Check if a url points to a valid location
+check_url() {
+    test $# == 1 && test "$1" || return 1
+    curl --output /dev/null --silent --head --fail "$1"
+}
+
+instantiate_url() {
+    test $# == 1 || exit
+    local url="${1}"
+    url=${url//\$\{arch\}/$(arch)}
+    url=${url//\$\{goarch\}/$(goarch)}
+    url=${url//\$\{goos\}/$RELEASE_GOOS}
+    url=${url//\$\{os\}/$RELEASE_OS}
+    url=${url//\$\{version\}/$(run_stoml version)}
+    echo "${url}"
+}
+
+# select either binary or tar download by checking for existence
+# and allow overriding with a local tool by setting the environment variable "$LOCAL_<tool>" (e.g. $LOCAL_wk)
+# (useful for testing and particularly useful on darwin since we don't publish a "wk" version for darwin)
+download_dependency() {
+  local tool="${1}"
+  local bin_dir="${2}"
+  local dependencies_toml="${DEP_FILE}"
+  local localToolVar="\$LOCAL_"${tool}
+  local localTool
+  localTool=$(eval "echo ${localToolVar}")
+
+    if [ -n "${localTool}" ]; then
+        cp "${localTool}" "${bin_dir}"
+        return 0
+    fi
+
+    run_stoml() {
+        local property="${1}"
+        "${bin_dir}"/stoml "${dependencies_toml}" "${tool}"."${property}"
+    }
+
+    local url_and_path
+    local fetch
+    local binarypath
+	local special_tarpath
+	local special_tarpath_url
+    binarypath=$(instantiate_url "$(run_stoml binarypath)")
+	special_tarpath=$(instantiate_url "$(run_stoml special_tarpath)")
+	special_tarpath_url=(${special_tarpath//;/ }) # split out special paths which contain <url>;<path in tarball>
+    local tarpath
+    tarpath=$(instantiate_url "$(run_stoml tarpath)")
+    echo $tarpath
```

Comment:
Would it be helpful to print the other paths/urls?  Maybe print them inside the if statements below?